odoo/modules/registry.py

```diff
@@ -677,17 +697,33 @@ def check_tables_exist(self, cr):
             for table in missing_tables:
                 _logger.error("Model %s has no table.", table2model[table])
 
-    def _clear_cache(self):
-        """ Clear the cache and mark it as invalidated. """
-        self.__cache.clear()
-        self.cache_invalidated = True
-
-    def clear_caches(self):
+    def clear_cache(self, *cache_names):
+        """ Clear the caches associated to methods decorated with
+        ``tools.ormcache``if cache is in `cache_name` subset. """
+        cache_names = cache_names or ('default',)
+        assert not any('.' in cache_name for cache_name in cache_names)
+        for cache_name in cache_names:
+            for cache in self.__caches_groups[cache_name]:
+                cache.clear()
+            self.cache_invalidated.add(cache_name)
+
+        # log information about invalidation_cause
+        caller_info = format_frame(inspect.currentframe().f_back)
+        log = _logger.info if self.loaded else _logger.debug
+        log('Invalidating %s model caches from %s', ','.join(cache_names), caller_info)
```

Comment:
What about logging with `stack_info=True` instead?
That's not a suggestion, I just want to know your feeling about this.

Reply:
Because it would be way to noisy.This is IMHO a useful information, but right now this is triggered quite a lot. I also had a version when a logged a few level of the stack, 2-3, but dropped it because it was still to noisy and most of the time we only see overrides of a crud method. I guess that this version is better than nothing, without adding multi line logs and staying discrete.